### PR TITLE
Fix color picker bug

### LINKED ISSUE #25013: Same background for multiple colorpickers
#### Describe the bug

Multiple color pickers will have the same background color even though they have different color values

#### Expected behavior

Each input must have a different background color
2021-06-18 at 1 21 AM

Actual

2021-06-18 at 1 21 AM

#### Steps to Reproduce

You will need a PrestaShop backoffice page which has multiple color pickers, I'm not aware of any page having that.
I discovered the bug on my module page

#### Additional information

* PrestaShop version: 1.7.8.0

### PR DESCRIPTION
**Pull request by unlocomqx**
When multiple pickers are in the page, they all will have the same background color even if they don't have the same color value

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | When multiple color pickers are in the page, they will all have the same background color even if they have different color values
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #25013
| How to test?      | You will need a page with multiple pickers, I used my module page as an exmple but couldn't find PrestaShop pages having multiple pickers

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25012)

### CODE REVIEW

**Comment by kpodemski** [comment]
I'm not sure if fixing it directly in third-party library is a way to go, maybe binding color picker should be done in the loop for every `mColorPicker` element?

**Comment by unlocomqx** [comment]
@kpodemski I tried that, but it had the same effect, because the color is updated for `$('.mColorPicker')` so it will take whatever color is in the closure and apply it to all `.mColorPicker` inputs. 
I fixed the third party lib because an edit on this lib caused the bug, it wasn't present in previous versions.
Btw, I managed to reproduce it on the "Order Notifications on the Favicon" module (a PrestaShop official module)

Before

2021-06-18 at 8 20 AM

After

2021-06-18 at 8 25 AM

File: js/jquery/plugins/jquery.colorpicker.js [hunk 1]
```diff
@@ -43,7 +43,7 @@
     });
     
     $(document).ready(function() {
-      $.fn.mColorPicker.setTextColor($('.mColorPicker'));
+     $.fn.mColorPicker.setTextColor($(this));
```
Comment (atomiix) [inline js/jquery/plugins/jquery.colorpicker.js]:
```suggestion
      $.fn.mColorPicker.setTextColor($(this));
```

Comment (unlocomqx) [reply js/jquery/plugins/jquery.colorpicker.js]: Reindented code

**Review by Progi1984 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Comment by atomiix** [comment]
@unlocomqx Thanks for your contribution! It would be great if you could provide a sample module to try it out so the QA team can validate your PR easily. Do you think you could do it?

**Comment by NeOMakinG** [comment]
> @unlocomqx Thanks for your contribution! It would be great if you could provide a sample module to try it out so the QA team can validate your PR easily. Do you think you could do it?

Hey @atomiix , @unlocomqx said that we could reproduce it with Order Notifications on the Favicon (a PS module)

**Review by NeOMakinG — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by unlocomqx** [comment]
> @unlocomqx Thanks for your contribution! It would be great if you could provide a sample module to try it out so the QA team can validate your PR easily. Do you think you could do it?

I included screenshots from a native module above

Thanks

**Review by PierreRambaud — DISMISSED** [review]

**Comment by hibatallahAouadni** [comment]
Hello @unlocomqx 

Thanks for your PR! It works:

image

But, I have a question, why in the multistore page **create a shop**, the color picker was white:

image

With your PR, it turns black:

image

Why? @NeOMakinG any idea?
Please check and feedback.

Thanks!

**Comment by unlocomqx** [comment]
@hibatallahAouadni I added a new commit to fix the issue

The reason was that the [code](https://github.com/PrestaShop/PrestaShop/pull/25012/commits/2ad154383045612360a408761af1944d12fcad3d#diff-2e3746771eaa16666e95c99394d3f07083d124a59889ddb861394c99198debbfR46) was updating the original color input, but that input has been removed at that point and replaced with the text input.
The solution is to keep track of the new inputs then update them correctly

**Review by NeOMakinG — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
Hello @unlocomqx 

LGTM, QA :heavy_check_mark: 
Thanks!

**Comment by Progi1984** [comment]
Thanks @unlocomqx &  @hibatallahAouadni

**Comment by PrestaEdit** [comment]
Hi all !

For informations, this issue was introduced inside the 1.7.8 PrestaShop version, it was working before.
Why does it reach the 8.0 version to be fixed, as this ?

**Comment by PrestaEdit** [comment]
And this is the Pull Request that introduce this: https://github.com/PrestaShop/PrestaShop/pull/20148

**Comment by Progi1984** [comment]
@PrestaShop/qa-functional Hi (or Good Night), could you confirm what @PrestaEdit says ? If it is the case, it should be a regression. Isn't it ?

**Comment by hibatallahAouadni** [comment]
Hello everyone,

The issue related used to be Bug and I don't know why it changed to Feature :sweat: 
That's why it was fixed on 8.0.0 and not on 1.7.8.0.
Well, I need to verify it if it's a regression on 1.7.8.0, cause with PS1767 and PS1778 I didn't reproduce the issue.

Thanks for your patience :pray:

**Comment by hibatallahAouadni** [comment]
@kpodemski @PrestaEdit @Progi1984 indeed it's a regression on PS1.7.8.0 and it should be fixed on 1.7.8.x.
Can we cherry-picked this PR to fix the issue on 1.7.8.x?

**Pull request merged**